packages/blocks/src/render-blocks.js

```diff
@@ -109,21 +109,26 @@ const appendInnerBlocks = ( blockName, elements, innerBlocks ) => {
  */
 const createElementsRecursive = ( elements ) => {
 	if ( isEmpty( elements ) ) {
-		return null;
+		return [];
 	}
 
 	return map( elements, ( element ) => {
 		if ( isString( element ) ) {
 			return element;
 		}
 
+		const children = [
+			...( element.innerBlocks || [] ),
+			...createElementsRecursive( element.children ),
+		];
+
 		return createElement(
 			element.name,
 			element.props,
 
 			// In practice, if an element is the host for inner blocks
 			// it's children should alwyas be empty here.
-			element.innerBlocks || createElementsRecursive( element.children )
```

Comment:
While debugging the code I found out that, for the block `core/cover` both `element.innerBlocks` and `element.children` are populated with some objects.

This is preventing the `<img />` from being added to the elements list.

Changing this solved the issue for the image but, if the Image Block has some text over it, the created `<p>` element is being added to the elements list before the `<img />` and, when rendered, the text gets hidden behind the image.

Took a look at WP.com behavior and there, the `core/cover` creates a wrapper with the class `wp-block-cover__inner-container` and adds the `<p>` element inside the wrapper.

Still didn't figure out why this is not happening on our side.